### PR TITLE
chore: Add debug info to JSX output

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Finding which file contains the component I want to edit has always been annoying to me (classic tedious solution: grepping codebase based on the text of the component) but I recently gleaned this cool solution to the problem from the console.dev newsletter: [LocatorJS](https://www.locatorjs.com/). It's a Chrome extension which takes you to the component's file _in your code editor_ on Option+click, which is ridiculously simple. It just needs JSX debug info to be emitted for file locations to be known.

## Changes

This upgrades `esbuild` to a version that supports the `jsxDev` setting (https://github.com/evanw/esbuild/releases/tag/v0.14.51) and enables it via `tsconfig.json` (`"jsx": "react-jsxdev"`). This way LocatorJS works beautifully.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,8 +95,8 @@
         "prop-types": "^15.7.2",
         "query-selector-shadow-dom": "^1.0.0",
         "rc-trigger": "^5.2.5",
-        "react": "^16.13.1",
-        "react-dom": "^16.12.0",
+        "react": "^16.14.0",
```
Comment (Twixes) [inline package.json:98]: We have already de facto been on 16.14.0 – https://github.com/PostHog/posthog/blob/a406f00aa02e3ae0a2a2db33cb330baf75ea3921/yarn.lock#L15578-L15579

File: tsconfig.dev.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": "./tsconfig.json",
+    "compilerOptions": {
+        "jsx": "react-jsxdev", // Support JSX in .tsx files, with debug info (e.g. file paths)
```
Comment (mariusandra) [inline tsconfig.dev.json:4]: `// ` is technically not valid `.json`. Hopefully the tools that'll ever read this file don't care.

**Review by mariusandra — APPROVED** [review]
Interesting... Give it a go!

**Comment by mariusandra** [comment]
Good luck with this though: 😅 
image

**Comment by Twixes** [comment]
Only had to update 50 manually after some targeted search-and-replace ;)

**Pull request merged**